AirCasting/CoreData/MeasurementStreamStorage.swift

```diff
@@ -269,6 +270,8 @@ final class HiddenCoreDataMeasurementStreamStorage: MeasurementStreamStorageCont
     func updateVersion(for sessionUUID: SessionUUID, to version: Int) throws {
         let sessionEntity = try context.existingSession(uuid: sessionUUID)
         sessionEntity.version = Int16(version)
+        // I needed to add this, because in the note editing function this get's called from another thread and the context save in `accessStorage` function was getting called before this function got exectuted.
+        try context.save()
```

Comment:
I think the error is on the client code side and should be fixed there rather than patched here.
The proper solution would be to add another `accessStorage` inside the **async callback** in our note editing functions like so (`deleteNote` function, but it needs to be done in other places too):
```Swift
measurementStreamStorage.accessStorage { [self] storage in
    do {
        try storage.deleteNote(note, for: sessionUUID)
        fetchSession(storage: storage) { session in
            self.sessionUpdateService.updateSession(session: session) { result in
                switch result {
                case .success(let updateData):
                    // HERE!
                    self.measurementStreamStorage.accessStorage {
                        try? $0.updateVersion(for: self.sessionUUID, to: updateData.version)
                        Log.info("Notes successfully updated")
                        completion()
                    }
                    //~~~
                case .failure(let error):
                    Log.info("Failed updating session while updating notes: \(error.localizedDescription)")
                    completion()
                }
            }
        }
    } catch {
        Log.info("Error when deleting note: \(error)")
    }
}
```